python/GafferTest/ContextTest.py

```diff
@@ -684,6 +684,66 @@ def testOmitCanceller( self ) :
 		context3 = Gaffer.Context( context1, omitCanceller = False )
 		self.assertIsNotNone( context3.canceller() )
 
+	@staticmethod
+	def collisionCountParallelHelper( seed, mode, chunk, countList ):
+		countList[0] = GafferTest.countContextHash32Collisions( 2**20, mode, chunk, seed )
+
+	@unittest.skipIf( GafferTest.inCI(), "Too expensive to run on CI, and has a small chance of failure because it is statistical" )
+	def testContextHashCollisions( self ) :
+
+		iters = 10
+		# Test all 4 context creation modes ( see ContextTest.cpp for descriptions )
+		for mode in [0,1,2,3]:
+			for chunk in [0,1,2,3]:
+				# This would be much cleaner once we can use concurrent.futures
+				import threading
+
+				results = [ [0] for i in range( iters ) ]
+				threads = []
+				for i in range( iters ):
+					x = threading.Thread( target=self.collisionCountParallelHelper, args=( i, mode, chunk, results[i] ) )
+					x.start()
+					threads.append( x )
+
+				for x in threads:
+					x.join()
+
+				s = 0
+				for l in results:
+					s += l[0]
+
+				# countContextHash32Collisions looks at just the lowest 32 bits of the hash.
```

Comment:
I thought we are looking at each of the four 32 bit chunks?

Reply:
Oh, yep, stale comment.

Reply:
Fixed in 717b9a8